Tests/SwiftSyntaxMacroExpansionTest/BodyMacroTests.swift

```diff
@@ -125,4 +125,30 @@ final class BodyMacroTests: XCTestCase {
       indentationWidth: indentationWidth
     )
   }
+
+  func testEmptyBodyMacro() {
+    struct EmptyBodyMacro: BodyMacro {
+      public static var formatMode: FormatMode { .disabled }
+
+      public static func expansion(
+        of node: AttributeSyntax,
+        providingBodyFor declaration: some DeclSyntaxProtocol & WithOptionalCodeBlockSyntax,
+        in context: some MacroExpansionContext
+      ) throws -> [CodeBlockItemSyntax] {
+        return []
+      }
+    }
+
+    assertMacroExpansion(
+      """
+      @EmptyBody
+      func f() {}
+      """,
+      expandedSource: """
+        func f() {}
```

Comment:
Is expected behavior is replacing the body in the original code to `{}`? If so, could you add some statements in the original source code?

Reply:
It’s expected to leave the body untouched. This matches the behavior of accessors

https://github.com/swiftlang/swift-syntax/blob/28794d2dc574b4d97aed235c43d852be2957d36e/Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift#L329

I updated the test.